### ISSUE TITLE
Crash when creating new world - 1.7/1.8

### ISSUE THREAD

**Issue by jimmyh77** [NOVA-Team/NOVA-Wrapper-MC#44]
I can recreate this every time by just downloading this template, doing "./gradlew idea" and then loading up idea and running the "17" client.  Also happens when running the 18 client

http://pastebin.com/y6VYvQyD

**Comment by calclavia** [comment]
That's weird, my workspace runs with no problem...

**Comment by RX14** [comment]
Run it in deobf.

**Comment by Dynious** [comment]
The weird thing is, it used to work before the '--refresh-dependencies' was fixed. This was on a NOVA Gradle dev env, so obf.

**Comment by calclavia** [comment]
I just replicated it using latest NOVA-Gradle with refresh dependencies. @RX14 pls take a look. :)

```
Time: 6/25/15 10:14 PM
Description: Exception in server tick loop

nova.core.nativewrapper.NativeManager$NativeException: NativeManager.toNova: Converter for net.minecraft.world.WorldServerMulti@5c4016e9 with class class net.minecraft.world.WorldServerMulti does not exist!
	at nova.core.nativewrapper.NativeManager.toNova(NativeManager.java:96)
	at nova.wrapper.mc18.launcher.ForgeEventHandler.worldUnload(ForgeEventHandler.java:20)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_12_ForgeEventHandler_worldUnload_Load.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:55)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:138)
	at net.minecraft.server.integrated.IntegratedServer.loadAllWorlds(IntegratedServer.java:102)
	at net.minecraft.server.integrated.IntegratedServer.startServer(IntegratedServer.java:126)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:438)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by RX14** [comment]
Go replicate it without NovaGradle. In MultiMC. I will bet you it's a issue with obf instead of NovaGradle.

Stop waiting for a fix and fix *your* shit.

**Comment by RX14** [comment]
Come up with a sensible reason as to why you suspect it is NovaGradle's fault before blaming it on me and asking me for a fix.

**Comment by calclavia** [comment]
@RX14 I found the issue. The NOVA API bundled in the development workspace is outdated/wrong. It gives this code (which is bugged and previously fixed).

```java
	private NativeConverter findConverter(Map<Class<?>, NativeConverter> map, Object obj) {
		Class<?> clazz = obj.getClass();
		NativeConverter nc = map.get(clazz);
		while (nc == null) {
			clazz = clazz.getSuperclass();
			if (clazz == Object.class) {
				return null;
			}
			nc = map.get(clazz);
		}
		return nc;
	}
```

But instead, it should have given the LATEST code:

```java
	private NativeConverter findConverter(Map<Class<?>, NativeConverter> map, Object obj) {
		Class<?> clazz = obj.getClass();

		return map
			.entrySet()
			.stream()
			.filter(e -> e.getKey().isAssignableFrom(clazz))
			.findFirst()
			.map(Map.Entry::getValue)
			.orElse(null);
	}

```

Seems like something wrong with Jenkins/Gradle/Maven not resolving the correct dep?

**Comment by RX14** [comment]
I would investigate if I had a computer. I would have been debugging this anyways if I had a computer.

**Comment by RX14** [comment]
Artifactory being strange, fixed by NOVA-Team/NOVA-Core@86ea02ee (run `gradle run17Client --refresh-dependencies`)
